apps/dotcom/fairy-worker/src/do/AgentDurableObject.ts

```diff
@@ -11,6 +12,47 @@ export class AgentDurableObject extends DurableObject<Environment> {
 		this.service = new AgentService(this.env)
 	}
 
+	private getUserDOStub(userId: string) {
+		const userDO = this.env.TL_USER.idFromName(userId)
+		return this.env.TL_USER.get(userDO)
+	}
+
+	private async getRateLimitError(
+		userId: string,
+		userStub: ReturnType<Environment['TL_USER']['get']>,
+		userIsAdmin: boolean
+	): Promise<Response | null> {
+		// Admins bypass rate limits
+		if (userIsAdmin) return null
+
+		const checkRes = await userStub.fetch(
+			`${INTERNAL_BASE_URL}/app/${userId}/fairy/check-rate-limit`,
+			{
+				method: 'POST',
+				headers: { 'Content-Type': 'application/json' },
+			}
+		)
```

Comment:
interesting that you opted for `fetch` instead of using rpc methods. did RPC not work?